### PR TITLE
John Papa Style Guide

### PR DESCRIPTION
**Pull request by RGFnaGFuIEd1bmF5**
I change whole project js and html files to john papa style guide. But its need more changes like modular structure and service provider.

### CODE REVIEW

**Comment by RGFnaGFuIEd1bmF5** [comment]
I removed modal editing. I think it should be situational implement. 
I moved modules out of entity folder.

File: app/templates/src/main/webapp/scripts/app/account/login/_login.controller.js [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
+        activate();
+        function activate() {
+
+        }
 
-        $scope.rememberMe = true;
         $timeout(function (){angular.element('[ng-model="username"]').focus();});
-        $scope.login = function (event) {
+        function login(event) {
+            console.log("Asd")
```
Comment (atomfrede) [inline app/templates/src/main/webapp/scripts/app/account/login/_login.controller.js]: I think you just forgot to remove the console output

**Comment by atomfrede** [comment]
:+1: Thanks! A lot of things to review I guess. For reference, the styleguide can be found here: https://github.com/johnpapa/angular-styleguide

**Comment by RGFnaGFuIEd1bmF5** [comment]
Actually there is lot of things to review thats right. I made it very quick. :) I will fix other things day by day.

File: app/templates/src/main/webapp/scripts/app/account/login/_login.controller.js [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
+        var vm = this;
+        vm.user = {};
+        vm.errors = {};
+        vm.rememberMe = true;
+        vm.login = login;
+
+        activate();
+        function activate() {
+
+        }
```
Comment (gzsombor) [inline app/templates/src/main/webapp/scripts/app/account/login/_login.controller.js]: I guess, they are not needed

**Comment by andidev** [comment]
+1 for this! Really nice to have a coding style to follow

**Comment by deepu105** [comment]
But this is too much refactoring to review and merge, i guess we should have discussed first and  come up with some stage by stage implementation,  this one has lot of merge conflicts and is tough to merge i guess.
@jdubois what do you think?

**Comment by mraible** [comment]
I agree this is great stuff! I actually referenced this in the JHipster mini-book. It'd be cool to say that JHipster implements it.

**Comment by deepu105** [comment]
ya I would also personally like to see constructor pattern etc as well so that when we migrate to angular 2 it will be easier, but I think it would be lot easier if its done in stages of small increments rather than single shot refactoring. This already has merge conflicts and will become worse day by day

**Comment by deepu105** [comment]
@RGFnaGFuIEd1bmF5 I went through the changelist, first let me say KUDOS :clap: thats hell lot of refactoring you have done there, its really great. but it would be a nightmare to merge this properly as this is too much, so can we be practical here and do something like below? of course again only if all of the dev team agrees!
Instead of merging this PR we can close it and then @RGFnaGFuIEd1bmF5 can open multiple smaller PRs as below so that its easier to review and merge
1. PR for entity js and html first as it will have most impact
2. PR for login, password, settings etc
3. PR for Metrics, health, logs, Audit etc
4. PR for any remaining items

What say @jdubois @atomfrede @mraible @gzsombor @gmarziou @andidev @cbornet @moifort

**Comment by mraible** [comment]
Sounds like a good plan to me. 👍

**Comment by atomfrede** [comment]
Yes, sounds easier to split this PR into smaller ones.

**Comment by andidev** [comment]
+1

**Comment by gmarziou** [comment]
+1

**Comment by deepu105** [comment]
@RGFnaGFuIEd1bmF5 also plz dont forget to test the following cases
1. App with default options and 1 or 2 entities in both dev and prod profile with grunt build
2. Same as above with gulp build
3. Same as 1 but with translation disabled

**Comment by tkman59** [comment]
+1

**Comment by jdubois** [comment]
+1 for me, this is a very important update we need to do!!

This PR is way too big -> we should have discussed this before (please read the guidelines), and we already have merge conflicts, so we probably can't merge this like this.

We should do some smaller commits and check each rule to see if what we do is correct. This is a lot of work!

**Comment by deepu105** [comment]
@RGFnaGFuIEd1bmF5 since all of us agree that this is a needed change but as smaller commits/prs can you close this and open small PRs as per above discussion, let us know if you need any help

**Comment by cbornet** [comment]
+1 of course

**Comment by RGFnaGFuIEd1bmF5** [comment]
Ok. Now I'm closing this PR and I opened an issue for smaller commits. #2271

File: app/templates/src/main/webapp/scripts/app/admin/health/_health.controller.js [hunk 3]
```diff
@@ -58,10 +83,9 @@ angular.module('<%=angularAppName%>')
...
             return result;
-        };
+        }
 
-
-        $scope.showHealth = function(health) {
+        function showHealth(health) {
             var modalInstance = $modal.open({
                 templateUrl: 'scripts/app/admin/health/health.modal.html',
                 controller: 'HealthModalController',
```
Comment (Vaelyr) [inline app/templates/src/main/webapp/scripts/app/admin/health/_health.controller.js]: vm missing.

File: app/templates/src/main/webapp/scripts/app/admin/health/_health.modal.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <div class="modal-header">
-    <button aria-label="Close" data-dismiss="modal" class="close" type="button" ng-click="cancel()"><span aria-hidden="true">&times;</span>
+    <button aria-label="Close" data-dismiss="modal" class="close" type="button" ng-click="vm.cancel()"><span aria-hidden="true">&times;</span>
     </button>
     <h4 class="modal-title" id="showHealthLabel"><% if (enableTranslation){ %>
 		{{'health.indicator.' + baseName(currentHealth.name) | translate}}<% }else{ %><span class="text-capitalize">{{ baseName(currentHealth.name) }}</span><% } %>
```
Comment (Vaelyr) [inline app/templates/src/main/webapp/scripts/app/admin/health/_health.modal.html]: baseName, currentHealth, subSystemName missing vm (also below)

File: app/templates/src/main/webapp/scripts/app/admin/health/health.html [hunk 5]
```diff
@@ -13,19 +13,19 @@ <h2 translate="health.title">Health Checks</h2>
...
                 <th class="col-md-2 text-center" translate="health.table.status">Status</th>
                 <th class="col-md-2 text-center" translate="health.details.details">Details</th>
-                
+
             </tr>
         </thead>
         <tbody>
-            <tr ng-repeat="health in healthData">
+            <tr ng-repeat="health in vm.healthData">
                 <td><% if (enableTranslation){ %>{{'health.indicator.' + baseName(health.name) | translate}}<% }else{ %><span class="text-capitalize">{{ baseName(health.name) }}</span><% } %> {{subSystemName(health.name)}}</td>
```
Comment (Vaelyr) [inline app/templates/src/main/webapp/scripts/app/admin/health/health.html]: baseName, currentHealth, subSystemName missing vm (maybe missed some)

**Pull request closed without merging**